aws/aws_eks/fargate/spot_and_fargate/main.tf

```diff
@@ -2,7 +2,7 @@
 
 # Documentation: https://www.terraform.io/docs/language/settings/index.html
 terraform {
-  required_version = ">= 0.14.0"
+  required_version = ">= 1.0.0"
```

Comment:
This example doesn't pass the `terraform apply` test:
* https://github.com/ContainerSolutions/terraform-examples/runs/3281538534?check_suite_focus=true#step:5:1924

```
aws_eks_node_group.changeme_spot_and_fargate_eks_node_group: Still creating... [24m20s elapsed]
╷
│ Error: Error creating EIP: AddressLimitExceeded: The maximum number of addresses has been reached.
│ 	status code: 400, request id: 06aa1004-23e4-4d15-ab27-73c19db3eaf7
│ 
│   with aws_eip.changeme_spot_and_fargate_eip[0],
│   on main.tf line 127, in resource "aws_eip" "changeme_spot_and_fargate_eip":
│  127: resource "aws_eip" "changeme_spot_and_fargate_eip" ***
│ 
╵
╷
│ Error: error waiting for EKS Node Group (eks-example-prod:changeme-eks-cluster-kube-system) to create: unexpected state 'CREATE_FAILED', wanted target 'ACTIVE'. last error: 1 error occurred:
│ 	* i-0231838c92f0e5a67: NodeCreationFailure: Instances failed to join the kubernetes cluster
│ 
│ 
│ 
│   with aws_eks_node_group.changeme_spot_and_fargate_eks_node_group,
│   on main.tf line 297, in resource "aws_eks_node_group" "changeme_spot_and_fargate_eks_node_group":
│  297: resource "aws_eks_node_group" "changeme_spot_and_fargate_eks_node_group" ***
│ 
╵
/home/runner/work/terraform-examples/terraform-examples/aws/aws_eks/fargate/spot_and_fargate
There are some problems with the CLI configuration:
╷
│ Error: Error parsing /home/runner/.terraformrc: At 3:16: illegal char
│
╵

As a result of the above problems, Terraform may not behave as intended.
```

Reply:
I've tested it in my AWS account and it worked fine.
Could this be related to some leaked AWS resources?

Checking the information here:
- https://github.com/hashicorp/terraform/issues/6018
- https://forums.aws.amazon.com/thread.jspa?threadID=254841

They say that: "By default, all AWS accounts are limited to 5 Elastic IP addresses per region". And we are using only 3 in this example.

Reply:
I've cleared the EIP limit but now hit next error in the CI:
* https://github.com/ContainerSolutions/terraform-examples/runs/3298936784?check_suite_focus=true#step:5:1847

```
aws_eks_node_group.changeme_spot_and_fargate_eks_node_group: Creation complete after 2m26s [id=eks-example-prod:changeme-eks-cluster-kube-system]
╷
│ Error: error waiting for Route in Route Table (rtb-0706f2a67e3bcdfc3) with destination (0.0.0.0/0) to become available: timeout while waiting for state to become 'ready' (timeout: 2m0s)
│ 
│   with aws_route.changeme_spot_and_fargate_route_private[2],
│   on main.tf line 168, in resource "aws_route" "changeme_spot_and_fargate_route_private":
│  168: resource "aws_route" "changeme_spot_and_fargate_route_private" ***
│ 
╵
/home/runner/work/terraform-examples/terraform-examples/aws/aws_eks/fargate/spot_and_fargate
```

Reply:
It's probably caused to the default timeout of 2m being too short.
The issue is described here:
* https://github.com/hashicorp/terraform/issues/8539
* https://github.com/hashicorp/terraform/issues/8542